lib/coordinates.rb

```diff
@@ -0,0 +1,12 @@
+class Coordinates
+  attr_reader :value, :grid
+
+  def initialize(value, grid)
+    @value = value
+    @grid = grid
+  end
+
+  def valid?
+    !grid.coordinate_invalid?(value)
```

Comment:
This expression is pretty hard to follow. Not invalid is a double negative. Does it need to be `coordinate_invalid?`?. If it were checking for valid, then this double negative wouldn't be needed.